### PR TITLE
Switch SAPM exporter to internal data model

### PR DESCRIPTION
**Pull request by dmitryax**
In addition:
- Align code to this change in core https://github.com/open-telemetry/opentelemetry-collector/commit/9d16f5971ef950b554d12db1533202e33f5b93c2.
- Fix SAPM exporter that was reporting incorrect dropped spans metric.

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
> * Align code to this change in core [open-telemetry/opentelemetry-collector@9d16f59](https://github.com/open-telemetry/opentelemetry-collector/commit/9d16f5971ef950b554d12db1533202e33f5b93c2).
> 
> * Fix SAPM exporter that was reporting incorrect dropped spans metric.

Can you please submit as 2 separate PRs?

File: exporter/sapmexporter/factory.go [hunk 1]
```diff
@@ -15,23 +15,24 @@
...
 	"github.com/open-telemetry/opentelemetry-collector/component"
 	"github.com/open-telemetry/opentelemetry-collector/config/configerror"
 	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
-	"go.uber.org/zap"
 )
 
 const (
 	// The value of "type" key in configuration.
-	typeStr = "sapm"
+	typeStr = configmodels.Type("sapm")
```
Comment (tigrannajaryan) [inline exporter/sapmexporter/factory.go:27]: This is not necessary and it will require less changes elsewhere if you keep `typeStr = "sapm"`.

**Review by tigrannajaryan — COMMENTED** [review]
LGTM, but please submit 2 separate PRs.

**Comment by dmitryax** [comment]
@tigrannajaryan submitted PR to update core dependency https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/192

Please take a look, I'll rebase this one after that

**Comment by dmitryax** [comment]
@tigrannajaryan rebased, PTAL

**Comment by dmitryax** [comment]
Perfomance test results: 

Before:
```
Test                                    |Result|Duration|CPU Avg%|CPU Max%|RAM Avg MiB|RAM Max MiB|Sent Items|Received Items|
----------------------------------------|------|-------:|-------:|-------:|----------:|----------:|---------:|-------------:|
Trace10kSPS/SAPM                        |PASS  |     15s|    27.3|    28.8|         52|         63|    148900|        148900|
```

After:
```
Test                                    |Result|Duration|CPU Avg%|CPU Max%|RAM Avg MiB|RAM Max MiB|Sent Items|Received Items|
----------------------------------------|------|-------:|-------:|-------:|----------:|----------:|---------:|-------------:|
Trace10kSPS/SAPM                        |PASS  |     15s|    23.2|    26.0|         52|         64|    149900|        149900|
```

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**